video/image_writer.c

```diff
@@ -262,7 +297,237 @@ static bool write_jpeg(struct image_writer_ctx *ctx, mp_image_t *image, FILE *fp
 
     jpeg_destroy_compress(&cinfo);
 
-    return true;
+    return !fclose(fp);
+}
+
+#endif
+
+#if HAVE_AVIF_MUXER
+
+static void log_side_data(struct image_writer_ctx *ctx, AVPacketSideData *data,
+                          int size)
+{
+    static char dbgbuff[129];
+    if (size)
+        MP_DBG(ctx, "write_avif() packet side data:\n");
+
+    for (int i = 0; i < size; i++) {
+        AVPacketSideData *sd = &data[i];
+        int k = 0;
+        for (; k < MPMIN(sd->size, 64); k++)
```

Comment:
I feel like there has to be a more efficient way to convert a buffer to a hex string, but if not, then so be it. I'm not sure if ffmpeg exposes these macros, but something like `AV_RB32()` followed by printf with `%08x` seems more efficient? I'm not sure.

Reply:
I'll just mp_msg_test for debug at the top of the function.